stdlib/LinearAlgebra/src/structuredbroadcast.jl

```diff
@@ -64,7 +64,24 @@ find_bidiagonal() = throw(ArgumentError("could not find Bidiagonal within broadc
 find_bidiagonal(a::Bidiagonal, rest...) = a
 find_bidiagonal(bc::Broadcast.Broadcasted, rest...) = find_bidiagonal(find_bidiagonal(bc.args...), rest...)
 find_bidiagonal(x, rest...) = find_bidiagonal(rest...)
+
+find_uplo(a::Char) = a
+find_uplo(a::Missing) = a
+function find_uplo(a::Bidiagonal, rest...)
+    if length(rest) > 0
+        return a.uplo == rest[1].uplo ? a.uplo : missing
```

Comment:
Here's a missing test case:

```julia
julia> Bidiagonal(1:5, 2:5, 'U') .* 2
ERROR: type Int64 has no field uplo
```

It's a little tricky because we actually need two missing value sentinels, I think.  One that represents the fact that we've not found _any_ `Bidiagonal` yet, and another that represents the fact that we've found incompatible uplos.  Or we'll need to make two passes through the Broadcasted expression; one that short-circuits to pull out the first `Bidiagonal`'s uplo and another that merges all uplos together, starting with that one.

Other nasty test-cases include nestings like `(1 .* 2) .* (Bidiagonal(1:3, 1:2, 'U') .* (3 .* 4)) .* (5 .* Bidiagonal(1:3, 1:2, 'L'))`

Reply:
@mbauman 8de30ec rewrote `find_uplo` it can now handle the edge cases:
```
julia> Bidiagonal(1:5, 2:5, 'U') .* 2
5×5 Bidiagonal{Int64,Array{Int64,1}}:
 2  4  ⋅  ⋅   ⋅
 ⋅  4  6  ⋅   ⋅
 ⋅  ⋅  6  8   ⋅
 ⋅  ⋅  ⋅  8  10
 ⋅  ⋅  ⋅  ⋅  10

julia> 2 .* Bidiagonal(1:5, 2:5, 'U')
5×5 Bidiagonal{Int64,Array{Int64,1}}:
 2  4  ⋅  ⋅   ⋅
 ⋅  4  6  ⋅   ⋅
 ⋅  ⋅  6  8   ⋅
 ⋅  ⋅  ⋅  8  10
 ⋅  ⋅  ⋅  ⋅  10

julia> (1 .* 2) .* (Bidiagonal(1:3, 1:2, 'U') .* (3 .* 4)) .* (5 .* Bidiagonal(1:3, 1:2, 'L'))
3×3 Tridiagonal{Int64,Array{Int64,1}}:
 120    0     ⋅
   0  480     0
   ⋅    0  1080
```

Also added test cases to the unit tests